src/vs/workbench/api/common/extHostNotebook.ts

```diff
@@ -322,6 +323,17 @@ export class ExtHostNotebookController implements ExtHostNotebookShape {
 			throw new Error('Document version mismatch');
 		}
 
+		if (!this._extHostFileSystem.value.isWritableFileSystem(uri.scheme)) {
+			throw new files.FileOperationError(localize('err.readonly', "Unable to modify read-only file '{0}'", this._resourceForError(uri)), files.FileOperationResult.FILE_PERMISSION_DENIED);
+		}
+
+		// validate write
+		const statBeforeWrite = await this._validateWriteFile(uri, options);
+
+		if (!statBeforeWrite) {
+			await this._mkdirp(uri);
```

Comment:
@rebornix `writeFile` should take care of creating parent folders as needed, any reason you do this anyway?